qiskit_experiments/calibration/experiments/rabi.py

```diff
@@ -0,0 +1,337 @@
+            duration=160,
+            sigma=40,
+        )
+
+    def __init__(self, qubit: int, amplitudes: Optional[Union[List[float], np.array]] = None):
+        """Setup a Rabi experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the Rabi experiment.
+            amplitudes: The amplitudes that will be scanned.
+        """
+        super().__init__([qubit])
+
+        if amplitudes is not None:
+            self._amplitudes = amplitudes
+        else:
+            self._amplitudes = np.linspace(-0.95, 0.95, 51)
+
+    def circuits(
+        self,
+        backend: Optional[Backend] = None,
+        schedule: Optional[ScheduleBlock] = None,
+        **circuit_options,
+    ) -> List[QuantumCircuit]:
+        """Create the circuits for the Rabi experiment.
+
+        Args:
+            backend: A backend object.
+            schedule: The schedule for which to scan the amplitude. This schedule must have
+                one parameter that will be scanned.
+            circuit_options: Circuit options that may include "amplitudes", i.e. the run-time
+                given amplitudes that will override those given be the init method.
+
+        Returns:
+            A ist of circuits with a rx rotation with a calibration whose amplitude is scanned.
+
+        Raises:
+            QiskitError:
+                - If the user-provided schedule does not contain a channel with an index
+                  that matches the qubit on which to run the Rabi experiment.
+                - If the user provided schedule has more than one free parameter.
+        """
+        if schedule is None:
+            amp = Parameter("amp")
+            with pulse.build() as default_schedule:
+                pulse.play(
+                    pulse.Gaussian(
+                        duration=self.experiment_options.duration,
+                        amp=amp,
+                        sigma=self.experiment_options.sigma,
+                    ),
+                    pulse.DriveChannel(self.physical_qubits[0]),
+                )
+
+            schedule = default_schedule
+        else:
+            if self.physical_qubits[0] not in set(ch.index for ch in schedule.channels):
+                raise QiskitError(
+                    f"User provided schedule {schedule.name} does not contain a channel "
+                    "for the qubit on which to run Rabi."
+                )
+
+        if len(schedule.parameters) != 1:
+            raise QiskitError("Schedule in Rabi must have exactly one free parameter.")
+
+        param = next(iter(schedule.parameters))
+
+        circuit = QuantumCircuit(1)
+        circuit.rx(param, 0)
+        circuit.measure_active()
+        circuit.add_calibration("rx", (self.physical_qubits[0],), schedule, params=[param])
+
+        circs = []
+        for amp in circuit_options.get("amplitudes", self._amplitudes):
+            assigned_circ = circuit.assign_parameters({param: amp}, inplace=False)
```

Comment:
Are you following @nkanazawa1989's suggestion for using parametrized circuits in our experiments? Or is it another usage of parametrized circuits, different from what @nkanazawa1989 was talking about?

Reply:
This is the same usage of parameter with spectroscopy.